### DISCUSSION TITLE
Systray quit function

### DISCUSSION THREAD

**Discussion by Dmitriy147** [fyne-io/fyne#4116] [General]
Need to use systray for multiservice app. Is it any way to handle  Quit event to make some functions (close working goroutines...) before exit?

**Comment by dweymouth** [comment]
If you just need to do some cleanup tasks you can do that without intercepting an event since the quit only exits the Fyne main loop, it doesn't do `os.Exit` on the whole application. So anything you put in `main` after the blocking `win.ShowAndRun()` call will run at that point.

**Comment by Dmitriy147** [comment]
Indeed! Right. Thanks! ..
